draft-pda-core.md

```diff
@@ -291,11 +291,437 @@ Each PA task in this document is divided into three sub-protocols as follows.
+each (leader, helper) pair? Probably not in general, but maybe for Prio?]
+
+The payload of the POST request is structured as follows:
+
+~~~
+struct {
+  PATask task;
+  opaque id[16];              // Equal to PAUploadStartReq.id
+  uint8 helper_key_config_id; // Euqal to HPKEKeyConfig.id
+  Url helper_url;             // URL of helper endpoint
+  PAHelperShare helper_share; // The helper's encrypted share
+  PALeaderShare leader_share; // The leader's share
+} PAUploadFinishReq;
+
+struct {
+  opaque enc<0..2^16-1>;     // Encapsulated HPKE context
+  opaque payload<0..2^16-1>; // The encrypted share (may be empty)
+} PAHelperShare;
+
+opaque PALeaderShare<0..2^16-1>;
+~~~
+
+This message is called a *report*.
+
+**Response.**
+The leader responds to well-formed requests to `[leader]/upload_finish` with
+status 200 and an empty payload. Malformed requests are handled as described in
+{{pa-error-common-aborts}}.
+
+## Verify {#pa-verify}
+
+[TODO: Add an illustration of this sub-protocol.]
+
+After the client uploads a report to the leader, the leader and helper verify
+the validity proof as specified by the PA protocol. This phase involves two
+requests sent from by the leader to the helper: the *verify start request*, and
+the *verify finish request*. The contents of each request depend largely on the
+specific PA protocol. However, both requests contain the helper's share of the
+report, allowing the helper to run the verification protocol statelessly.
+
+Another important feature of the verification protocol is that the leader may
+"batch" multiple runs of the protocol into the same message flow. This allows
+the leader to throttle the load on the helper endpoint.
+
+At the end of this phase, the leader and helper will have decided whether a set
+of client inputs are valid. For each valid input, they proceed as described in
+{{pa-collect}}.
+
+### Verify Start
+
+The leader begins the protocol by collecting a sequence of helper shares that
+all correspond to the same helper endpoint URL, key configuration, and PA task.
+For each share, it generates the first protocol messages as specified by the PA
+protocol, then proceeds as follows.
+
+**Request.**
+Let `[helper]` denote the helper URL. The leader sends a POST request to
+`[helper]/verify_start` with the following payload:
+
+~~~
+struct {
+  PATask task;
+  uint8 key_config_id;
+  PAHelperShare shares<0..2^24-1>;       // Any number of helper shares
+  select (PAVerifyStartReq.task.proto) { // Payload for each helper share
+    case prio: PrioVerifyStartReq payloads<0..2^24-1>;
+    case hits: HitsVerifyStartReq payloads<0..2^24-1>;
+  }
+} PAVerifyStartReq;
+~~~
+
+Each helper share has a corresponding message payload contained in
+`PAVerifyStartReq.payloads`. This includes any protocol-specific information the
+helper needs for the first step of input validation, e.g., the joint randomness
+used by the leader and helper for the protocol run. Note that a well-formed
```

Comment:
We've not defined joint randomness, so I might drop this example text entirely. The preceding text seems clear enough as-is.